source/docs/flow_control.md

```diff
@@ -101,6 +101,29 @@ upstream connections, the `readDisable(true)` calls are unwound in
 ClientConnectionImpl::onMessageComplete() to make sure that as connections are
 returned to the connection pool they are ready to read.
 
+#### HTTP2 defer processing backed up streams
+
+This section will be further integrated with the rest of the document when it is
+enabled by default (currently off by default). At a high level this change does
+the following:
+
+* If a HTTP/2 stream is read disabled anywhere we will begin buffering body and
+  trailers in the receiving side of the codec to minimize work done on the
+  stream when it's read disabled. There will be a callback scheduled to drain
+  these when the stream is read enabled, it can also be drained if the stream is
+  read enabled, and the stream is invoked with additional data.
+* The codec's recieve buffer high watermark is still used in consideration for
+  granting peer stream additional window (preserving existing protocol flow
+  control), but no longer read disables the stream, as we could be read disabled
+  elsewhere, buffer data in codec's recieve buffer hitting high watermark and
```

Comment:
Per the giant comment thread, my understanding also was that the recv side read disable was never really used, as the data was always sent directly through. I remembering commenting on this eons ago with @alyssawilk originally did the work. I think this is what @yanavlasov was referring to also. Assuming this is the case, can you pedantically clarify this statement just to make it more clear that we were never previously actually relying on this? I think the idea now is that we still do read disable "from the other side" we just now buffer on the origin side vs. the receiving side. Right?

Reply:
Clarified as suggested. The idea now is we read disable from the other side (as done prior), but buffer on the receiving side vs the connection output buffer which the data would get serialized to.